Adjust Keda docs according to the documentation requirements (#156)
kyma-project/keda-manager: README.md

# Keda module

## Overview 

Keda module consists of Keda Manager that is an extension to the Kyma runtime. It allows users to install KEDA. It follows the Kubernetes operator pattern to manage the lifecycle of the KEDA installation based on the existence and the content of the dedicated Keda custom resource (CR).

> ![Keda module overview](./docs/assets/keda-overview.drawio.svg)

Reviewer: (1) User applies Keda Custom Resource
(2) Keda manager watches Keda Kustom Resource
(3) Keda manager  reconciles keda workloads

# Keda module

## Overview 

Keda module consists of Keda Manager that is an extension to the Kyma runtime. It allows users to install KEDA. It follows the Kubernetes operator pattern to manage the lifecycle of the KEDA installation based on the existence and the content of the dedicated Keda custom resource (CR).

![Keda module overview](./docs/assets/keda-overview.drawio.svg)

1. User applies the Keda CR.
2. Keda Manager watches the Keda CR.
3. Keda Manager reconciles the KEDA workloads.

> For more information, see [Use Keda Manager to manage KEDA](/docs/keda-management.md).

Reviewer: wrong link

> # Keda module

Author: I wasn't sure. They are not capitalized on our website. For example, check [here](https://kyma-project.io/docs/kyma/latest/02-get-started/01-quick-install/).

> # Keda module

Reviewer 2: You're absolutely right -> https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/03-formatting.md#headings. We capitalize them in DITA CCMS, right?

# Keda module

## Overview 

Keda module consists of Keda Manager that is an extension to the Kyma runtime. It allows users to install KEDA. It follows the Kubernetes operator pattern to manage the lifecycle of the KEDA installation based on the existence and the content of the dedicated Keda custom resource (CR).

![Keda module overview](./docs/assets/keda-overview.drawio.svg)

1. User applies the Keda CR.
2. Keda Manager watches the Keda CR.
3. Keda Manager reconciles the KEDA workloads.

For more information, see [Use Keda Manager to manage KEDA](docs/contributor/02-10-management.md).

### What is KEDA?

KEDA is a flexible Event Driven Autoscaler for the Kubernetes workloads. It extends the Kubernetes autoscaling mechanisms with its own metric server and the possibility to make use of external event sources for making scaling decisions. To learn more about KEDA, see the [KEDA documentation](https://keda.sh/docs/latest/concepts/).

> ## Read More

Reviewer 2: ```suggestion
## Read more
```
sentence case?

If you want to use Kyma's Keda module, check the [user](/docs/user/) folder to learn more about it. In this folder, you can also find information on how to configure your module. You can also find a demo application that shows how to scale the Kubernetes workloads using Keda API.

> The [contributor](/docs/contributor/) folder includes all the necessary information you may need if you want to extend the module with new features. You can learn more about the project structure, make targets, CI/CD jobs that are part of the developing cycle, and different installation options.

Reviewer 2: ```suggestion
The [contributor](/docs/contributor/) folder includes all the necessary information on how to extend the module with new features. You can learn more about the project structure, make targets, CI/CD jobs that are part of the developing cycle, and different installation options.
```

> # Keda Manager

Reviewer 2: I thought the documentation in `.docs` should relate to the Keda Module and Keda Manager is a part of the module. 
```suggestion
# Keda Module
```

> **NOTE:** For more detailed documentation, check the [`docs`](/docs/README.md) folder.

## Overview 

> Keda Manager is an extension to the Kyma ecosystem that allows users to install KEDA. It follows the Kubernetes operator pattern to manage the lifecycle of the KEDA installation based on the existence and the content of the dedicated Keda custom resource (CR).

Reviewer 3: the diagram does not conform to our [diagram guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/09-diagrams.md).

# Keda module

## Overview 

Keda module consists of Keda Manager that is an extension to the Kyma runtime. It allows users to install KEDA. It follows the Kubernetes operator pattern to manage the lifecycle of the KEDA installation based on the existence and the content of the dedicated Keda custom resource (CR).

![Keda module overview](./docs/assets/keda-overview.drawio.svg)

1. User applies the Keda CR.
2. Keda Manager watches the Keda CR.
3. Keda Manager reconciles the KEDA workloads.

For more information, see [Use Keda Manager to manage KEDA](docs/contributor/02-10-management.md).

### What is KEDA?

KEDA is a flexible Event Driven Autoscaler for the Kubernetes workloads. It extends the Kubernetes autoscaling mechanisms with its own metric server and the possibility to make use of external event sources for making scaling decisions. To learn more about KEDA, see the [KEDA documentation](https://keda.sh/docs/latest/concepts/).

## Read more

> If you want to use Kyma's Keda module, check the [user](/docs/user/) folder to learn more about it. In this folder, you can also find information on how to configure your module. You can also find a demo application that shows how to scale the Kubernetes workloads using Keda API.

Reviewer 3: drop the "can"

```suggestion
If you want to use Kyma's Keda module, check the [user](/docs/user/) folder to learn more about it. In this folder, you also get information on how to configure your module. You also find a demo application that shows how to scale the Kubernetes workloads using Keda API.
```

# Keda module

## Overview 

Keda module consists of Keda Manager that is an extension to the Kyma runtime. It allows users to install KEDA. It follows the Kubernetes operator pattern to manage the lifecycle of the KEDA installation based on the existence and the content of the dedicated Keda custom resource (CR).

> ![a](./docs/assets/keda-overview.drawio.svg)

Reviewer 2: The diagram is not formatted according to [our guidelines](https://github.com/kyma-project/community/blob/main/docs/guidelines/content-guidelines/09-diagrams.md).

kubectl apply -f https://github.com/kyma-project/keda-manager/releases/latest/download/keda-manager.yaml
```

> 2. To get KEDA installed, apply the sample Keda CR:

Author: That's what I wanted to do, but I got [feedback](https://github.com/kyma-project/keda-manager/pull/100#discussion_r1113059392) that's it better to use the passive voice here as it's not the user who installs KEDA

2. To get KEDA installed, apply the sample Keda CR:

```bash
kubectl apply -f config/samples/operator_v1alpha1_keda_k3d.yaml
```

> For more installation options, check the [Install Keda Manager](/docs/keda-installation.md) tutorial.

Reviewer: wrong link

Current versions per each channel are represented by the [ModuleTemplate CR](https://github.com/kyma-project/lifecycle-manager/blob/main/docs/technical-reference/api/moduleTemplate-cr.md) submitted to a matching folder in the Kyma Git repository.

After you merge all the changes into the `main` branch in the `keda-manager` repository, the [CI/CD jobs](/docs/contributor/ci-cd.md) bundle module images and generate the ModuleTemplate for you.
Submit your ModuleTemplate into the desired channel using a pull request to the Kyma repository.
A series of governance jobs tests if the new candidate version fulfills the criteria described in the [module submission process](https://github.com/kyma-project/community/tree/main/concepts/modularization#module-submission-process).

> ## Keda module footprint

Reviewer: This part is also relevant to the user. Lets move it to the /user folder.

Instead of TBD, lets add the measured values per keda workload:

Keda module consists of four workloads in total.
Two if those ( `keda-manager` and `keda-operator`) are implementing kubernetes operator pattern and consume extra resources only when they detect changes in the resources they watch (i.e Keda CR, ScaledObject CR, etc.. ). They usually are idle and consume as little as few milicores of CPU time and less than 30MB of memory. At the time of active reconciliation the observed CPU time jumps to `5m`.

Similarly to the operators, `keda-admission-webhooks` stays idle most of the time and performs validation operations only when new object ( form `*.keda.sh` api group) are submitted by the user.

The last workload `keda-operator-metrics-apiserver` continuously serves metrics for the kubernetes autoscaling components. Here, the consumption is the highest, but in case for one or two active keda scalers it stays at 5-7 milicores of CPU time.

```

| Name                            | CPU (cores) | Memory (bytes) |
|---------------------------------|-------------|----------------|
| keda-admission-webhooks         | 1m          | 10Mi           |
| keda-manager                    | 3m          | 23Mi           |
| keda-operator                   | 3m          | 26Mi           |
| keda-operator-metrics-apiserver | 5m          | 30Mi           |
```

Current versions per each channel are represented by the [ModuleTemplate CR](https://github.com/kyma-project/lifecycle-manager/blob/main/docs/technical-reference/api/moduleTemplate-cr.md) submitted to a matching folder in the Kyma Git repository.

> Having merged all the changes into the `main` branch in the `keda-manager` repository, the [CI/CD jobs](/docs/CI-CD-description.md) will bundle module images and generate ModuleTemplate for you.

Reviewer 3: ```suggestion
Having merged all the changes into the `main` branch in the `keda-manager` repository, the [CI/CD jobs](/docs/CI-CD-description.md) bundles module images and generates a ModuleTemplate for you.
```

or 

```suggestion
Having merged all the changes into the `main` branch in the `keda-manager` repository, the [CI/CD jobs](/docs/CI-CD-description.md) bundles module images and generates the ModuleTemplate for you.
```

"Having merged" is such an impersonal phrase. Who or what does the merging? Can we use active voice with a subject in this sentence?

> Submit your ModuleTemplate into the desired channel using a pull request to the Kyma repository.

Reviewer 3: I'm not sure who is the audience of this instruction.
Is this step done by the end user who installs the Keda Manager on their cluster?
Or is this document for the Keda development team?

> A series of governance jobs will start testing if the new candidate version fulfills the criteria described in the [module submission process](https://github.com/kyma-project/community/tree/main/concepts/modularization#module-submission-process).

Reviewer 3: ```suggestion
A series of governance jobs tests if the new candidate version fulfills the criteria described in the [module submission process](https://github.com/kyma-project/community/tree/main/concepts/modularization#module-submission-process).
```
